### PR TITLE
Fix Create and Delete for CampaignContribution/CampaignRating

### PR DESCRIPTION
**Pull request by cungbac**
1. CampaignContribution
- Add CreateRequestValidator
- Add user check in delete service (if user is not the owner of the Campaign Contribution, then cannot delete)

2. CampaignRating
- Add CreateRequestValidator
- Add user check in delete service (if user is not the owner of the Campaign Raing, then cannot delete)

### CODE REVIEW

File: src/DonateHope.Core/Services/CampaignContributionServices/CampaignContributionDeleteService.cs [hunk 1]
```diff
@@ -38,6 +38,16 @@ string reasonForDeletion
         }
         
         var deletedCampaignContribution = queryResult.Value;
+        
+        if (deletedBy != deletedCampaignContribution.UserId)
```
Comment (AnNguyenLe) [inline src/DonateHope.Core/Services/CampaignContributionServices/CampaignContributionDeleteService.cs:48]: What is the business logic behind this condition?

Comment (AnNguyenLe) [reply src/DonateHope.Core/Services/CampaignContributionServices/CampaignContributionDeleteService.cs:48]: should we place the business logic behind if (deletedCampaignContribution.IsDeleted) for less confusing?

File: src/DonateHope.Core/Services/CampaignRatingServices/CampaignRatingDeleteService.cs [hunk 2]
```diff
@@ -35,6 +35,17 @@ Guid deletedBy
...
+        if (deletedBy != deletedCampaignRating.UserId)
+        {
+            _logger.LogWarning(
+                "User id {UserId} is unauthorized to update campaign contribution {CampaignRatingId}",
+                deletedBy,
+                deletedCampaignRating.Id
+            );
+            return new ProblemDetailsError("You are unauthorized to update this campaign rating.");
+        }
+
```
Comment (AnNguyenLe) [inline src/DonateHope.Core/Services/CampaignRatingServices/CampaignRatingDeleteService.cs:54]: My comment above applied here as well

File: src/DonateHope.Core/Services/CampaignContributionServices/CampaignContributionDeleteService.cs (hunk 1 above)
Comment (cungbac) [reply src/DonateHope.Core/Services/CampaignContributionServices/CampaignContributionDeleteService.cs:48]: Reordered the logic after checking isDeleted.
Please help to review again

**Review by AnNguyenLe — APPROVED** [review]

**Pull request merged**